src/backend/distributed/commands/multi_copy.c

```diff
@@ -598,7 +605,8 @@ CopyToExistingShards(CopyStmt *copyStatement, char *completionTag)
 		/* close the COPY input on all shard placements */
 		EndRemoteCopy(connectionList, true);
 
-		if (MultiShardCommitProtocol == COMMIT_PROTOCOL_2PC)
+		if (MultiShardCommitProtocol == COMMIT_PROTOCOL_2PC ||
+			cacheEntry->replicationModel == REPLICATION_MODEL_2PC)
```

Comment:
@mtuncer I suggest we leave it for the moment. @mtuncer is working on refactoring COPY to use the new connection API. At that point we should call the appropriate 2PC function and remove this code.